docs/operational_guide/availability_consistency_durability.md

```diff
@@ -0,0 +1,126 @@
+# Tuning Availability, Consistency, and Durability
+
+## Overview
+
+M3DB is designed as a High Availability [HA](https://en.wikipedia.org/wiki/High_availability) system because it doesn't use a consensus protocol like Raft or Paxos to enforce strong consensus and consistency guarantees.
+However, even within the category of HA systems, there is a broad spectrum of consistency and durability guarantees that a database can provide.
+To address as many use cases as possible, M3DB can be tuned to achieve the desired balance between performance, availability, durability, and consistency.
+
+Generally speaking, [the default and example configuration for M3DB](https://github.com/m3db/m3/tree/master/src/dbnode/config) favors performance and availability, as that is well-suited for M3DB's most common metrics and Observability use cases. To instead favor consistency and durability, consider tuning values as described in (TODO).
+Database operators who are using M3DB for workloads that require stricter consistency and durability guarantees should consider tuning the default configuration to better suite their use case.
+The rest of this document describes the various configuration options that are available to M3DB operators to make those tradeoffs.
+
+While reading the rest of this document, we recommend referring to [the default configuration file](https://github.com/m3db/m3/blob/master/src/dbnode/config/m3dbnode-all-config.yml) (which has every possible configuration value set) to see how the described values fit into M3DB's configuration as a whole.
+
+## Tuning for Performance and Availability
+
+### Client Write and Read consistency
+
+We recommend running the client with `writeConsistencyLevel` set to `majority` and `readConsistencyLevel` set to `unstrict_majority`.
+This means that all write must be acknowledged by a quorums of nodes in order to be considered succesful, and that reads will attempt to achieve quorum, but will return the data from a single node if they are unable to achieve quorum.
+You can read about the consistency levels in more detail in [the Consistency Levels section](../m3db/architecture/consistencylevels.md)
+
+### Commitlog Configuration
+
+We recommend running M3DB with an asynchronous commitlog.
+This means that writes will be reported as successful by the client, though the data may not have been flushed to disk yet.
+
+For example, consider the default configuration:
+
+```
+commitlog:
+  flushMaxBytes: 524288
+  flushEvery: 1s
+  queue:
+    calculationType: fixed
+    size: 2097152
+```
+
+This configuration states that the commitlog should be flushed whenever either of the following is true:
+
+1. 524288 or more bytes have been written since the last time M3DB flushed the commitlog.
```

Comment:
Why `524288` and `2097152`?

Reply:
its just the default we've always used. Presumably @robskillington did some benchmarking?

Reply:
they're 2^19 and 2^21, respectively, if that helps.